src/highdicom/legacy/sop.py

```diff
@@ -417,66 +545,1911 @@ def __init__(
+            "PertinentOtherEvidenceSequence",
+            "CompletionFlag",
+            "CompletionFlagDescription",
+            "VerificationFlag",
+            "PreliminaryFlag",
+            "IdenticalDocumentsSequence"
+         ]
+        }
+        self.earliest_date = DA('00010101')
+        self.earliest_time = TM('000000')
+        self.earliest_date_time = DT('00010101000000')
+        self.farthest_future_date = DA('99991231')
+        self.farthest_future_time = TM('235959')
+        self.farthest_future_date_time = DT('99991231235959')
+        self._slices: list = []
+        self._tolerance = 0.0001
+        self._slice_location_map: dict = {}
+        self._byte_data = bytearray()
+        self._word_data = bytearray()
+        self.earliest_content_date_time = self.farthest_future_date_time
+        self._add_common_ct_pet_mr_build_blocks()
+        if (_SOP_CLASS_UID_IOD_KEY_MAP[sop_class_uid] ==
+                'legacy-converted-enhanced-ct-image'):
+            self._add_build_blocks_for_ct()
+        elif (_SOP_CLASS_UID_IOD_KEY_MAP[sop_class_uid] ==
+                'legacy-converted-enhanced-mr-image'):
+            self._add_build_blocks_for_mr()
+        elif (_SOP_CLASS_UID_IOD_KEY_MAP[sop_class_uid] ==
+                'legacy-converted-enhanced-pet-image'):
+            self._add_build_blocks_for_pet()
+
+    def _is_empty_or_empty_items(self, attribute: DataElement) -> bool:
+        """Takes a dicom DataElement and check if DataElement is empty or in
+        case of Sequence returns True if there is not item or all the items
+        are empty.
+        """
+        if attribute.is_empty:
+            return True
+        if isinstance(attribute.value, DataElementSequence):
+            if len(attribute.value) == 0:
+                return True
+            for item in attribute.value:
+                for tg, v in item.items():
+                    v = item[tg]
+                    if not self._is_empty_or_empty_items(v):
+                        return False
+        return False
+
+    def _mark_tag_as_used(self, tg: BaseTag) -> None:
+        """Checks what group the input tag belongs to and marks it as used to
+        keep track of all used and unused tags
+        """
+        if tg in self._shared_tags:
+            self._shared_tags[tg] = True
+        elif tg in self.excluded_from_perframe_tags:
+            self.excluded_from_perframe_tags[tg] = True
+        elif tg in self._perframe_tags:
+            self._perframe_tags[tg] = True
+
+    def _copy_attrib_if_present(self, src_ds: Dataset, dest_ds: Dataset,
+                                src_kw_or_tg: str, dest_kw_or_tg: str = None,
+                                check_not_to_be_perframe: bool = True,
+                                check_not_to_be_empty: bool = False) -> None:
+        """Copies a dicom attribute value from a keyword in the source Dataset
+        to a similar or different keyword in the destination Dataset
+        Parameters
```

Comment:
Couple of points regarding docstrings:

* Keep space between sections and at the end of the docstring
* Use the absolute namespace for documentation of custom types from external packages, e.g., `pydicom.dataset.Dataset` instead of `Dataset`
* Document type of return value (not relevant to this particular method)